Fix incomplete thirdparty guidelines on Complying with licenses (#8513)
godotengine/godot-docs: about/complying_with_licenses.rst

Complying with licenses
=======================

.. warning::

    The recommendations in this page **are not legal advice.** They are provided
    in good faith to help users navigate license attribution requirements.

What are licenses?
------------------

Godot is created and distributed under the `MIT License <https://opensource.org/licenses/MIT>`_.

> It doesn't have a sole owner either, as every contributor that submits code to

Reviewer: ```suggestion
It doesn't have a sole owner, as every contributor that submits code to
```

    If you are interested in licence compliance as a contributor, you can find
    guidelines :ref:`here <doc_best_practices_for_engine_contributors_license_compliance>`.

.. tip::

    Alongside the Godot license text, remember to also list third-party notices
    for assets you're using, such as textures, models, sounds, music and fonts.

>     Free assets in particular often come with licenses that require attribution.

Reviewer: I wonder whether this line is necessary, as in, why particularly highlight free assets? I think all assets used should be checked. It seems to me that the previous paragraph ("remember to also list third-party notices") is all that's required here.

Reviewer 2: See the line below and it becomes obvious 🙂

Author: I'll rephrase a bit.

    Copyright (c) 2007-2014 Juan Linietsky, Ariel Manzur.

    Permission is hereby granted, free of charge, to any person obtaining a copy
    of this software and associated documentation files (the "Software"), to deal
    in the Software without restriction, including without limitation the rights
    to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
    copies of the Software, and to permit persons to whom the Software is
    furnished to do so, subject to the following conditions:

    The above copyright notice and this permission notice shall be included in all
    copies or substantial portions of the Software.

    THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
    IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
    FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
    AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
    LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
    OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
    SOFTWARE.

> Beside its own MIT license, Godot includes code from a number of third party

Reviewer: ```suggestion
Beside its own MIT license, Godot includes code from a number of third-party
```

In the case of the MIT license, the only requirement is to include the license
text somewhere in your game or derivative project.

> This text reads as follows::

Author: Makes it a code block, so that it's not extracted for translation.

Reviewer: Hmm, but with the added line breaks, that makes it slightly more annoying to reuse in a project...

Author: I'd say it's a necessary evil in this case :)

    - :ref:`Engine.get_copyright_info<class_Engine_method_get_copyright_info>`

.. _doc_complying_with_licenses_thirdparty:

Third-party licenses
--------------------

Godot itself contains software written by
`third parties <https://github.com/godotengine/godot/blob/master/thirdparty/README.md>`_,
which is compatible with, but not covered by Godot's MIT license.

Many of these dependencies are distributed under permissive open source licenses
which require attribution by explicitly citing their copyright statement and
license text in the final product's documentation.

Given the scope of the Godot project, this is fairly difficult to do thoroughly.
For the Godot editor, the full documentation of third-party copyrights and
licenses is provided in the `COPYRIGHT.txt <https://github.com/godotengine/godot/blob/master/COPYRIGHT.txt>`_
file.

A good option for end users to document third-party licenses is to include this
file in your project's distribution, which you can e.g. rename to
``GODOT_COPYRIGHT.txt`` to prevent any confusion with your own code and assets.

If you export your project using a
:ref:`custom build with specific modules disabled <doc_optimizing_for_size>`,

> you can remove the copyright lines for the libraries you compiled out.

Reviewer 2: ```suggestion
you can remove the copyright lines for the libraries you disabled.
```
I can't make sense of "compiled out" as a phrase

Author: I removed the paragraph as a whole in the end.
